code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -72,6 +72,19 @@
 	return ..()
 
 /mob/living/simple_animal/hostile/Life()
+	// Indeed, making the mobs NOT act speeds the process up
```

Comment:
Recommend to put a cooldown on checking for players, to reduce the number of times we run this loop, and to add a bit of chase to the mobs.

recommend var/active and a var/playerCheck as bool values in the /hostile class. Also shift this new chunk into its own proc of nearPlayers() or something, that returns true or false. Might be other instances we want to set it off manually, like on new() or something.


```
/proc/nearPlayers()
	var/list/Mobs = ohearers(vision_range, targets_from)
	for(var/mob/M in Mobs)
		if(M.ckey)
			if(!findtext(M.ckey,"@"))
				return true
	//would have returned by now if we found something, so return false because we didnt
	return false
```

and then in the life() proc

```
proc/life()
<anything that needs to happen, regardless of active, in life()>
//start check for players
if(playerCheck)
	active = nearPlayers() //store if we are active or not
	playerCheck = 0
	spawn(1500) //wait 1.5 seconds, (or whatever the conversion is) then check again
		playerCheck = 1
//end check for players
//check if we are active since last playercheck
if(!active)
	return
<rest of life() proc>
```